src/NexusMods.Common/GuidedInstaller/ValueObjects/AssetUrl.cs

```diff
@@ -1,10 +1,14 @@
+using JetBrains.Annotations;
 using Vogen;
 
-namespace NexusMods.Common.UserInput;
+namespace NexusMods.Common.GuidedInstaller.ValueObjects;
+
+// TODO: convert this into a discriminated union using OneOf
```

Comment:
What do you mean here? What would the alternatives of the union be?

Reply:
> URL of an asset that's tied to an option. This can be a relative path, or a URL to a remote resource.

That is the description of `AssetUrl`. It can be:

1) a relative path
2) a URL

As such, you could model this type as a union: `AssetUrl : OneOfBase<RelativePath, URI>`.

Reply:
Ah, I understand now. I missed the RelativePath option. Yeah a OneOf would make this much clearer